src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer_ILspLogger.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.LanguageServer;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.LogHub;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient
+{
+    internal partial class InProcLanguageServer : ILspLogger
+    {
+        public async Task<TraceSource?> CreateTraceSourceAsync(string logName, CancellationToken cancellationToken)
+        {
+            if (_asyncServiceProvider == null)
+                return null;
+
+            var cleaned = string.Concat(logName.Split(Path.GetInvalidFileNameChars(), StringSplitOptions.RemoveEmptyEntries));
+            var logId = new LogId(cleaned, new ServiceMoniker("Microsoft.VisualStudio.LanguageServices.Implementation.LanguageClient"));
+
+            var serviceContainer = await _asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>().ConfigureAwait(false);
+            var service = serviceContainer.GetFullAccessServiceBroker();
+
+            using var configuration = await TraceConfiguration.CreateTraceConfigurationInstanceAsync(service, cancellationToken).ConfigureAwait(false);
+
+            var traceSource = await configuration.RegisterLogSourceAsync(logId, new LogHub.LoggerOptions(), cancellationToken).ConfigureAwait(false);
+            traceSource.Switch.Level = SourceLevels.ActivityTracing | SourceLevels.Verbose;
```

Comment:
Verbose? Really? That could be really slow. And isn't the idea to let LogHub set the switch level?

Reply:
That doesn't help our core case, which is to run into issues after the fact, and then want to report an issue that includes all the logs up to that point.

Reply:
siwtched to informational level logging.

Reply:
Many people are deciding they want logging for this purpose. This is okay, however I'll give you the same warning as everyone else: you own the perf impact here caused by the logging.

Reply:
You also may want to make it `SourceLevels.Informational` because that's the most detailed level you use.